vignettes/tutorial/resample.Rmd

```diff
@@ -305,16 +305,84 @@ r = resample("regr.rpart", bh.task, rdesc, show.info = FALSE)
+To initiate the grouping, we need to set `blocking.cv = TRUE` when creating the resample description object.
+
+```{r}
+rdesc = makeResampleDesc("CV", iters = 3, blocking.cv = TRUE)
+p = resample("classif.lda", task, rdesc)
+
+sort(p$pred$instance$train.inds[[1]])
+```
+
+However, please note the effects of this method: The created folds will not have the same size!
+Here, Fold 1 has a 120/30 split while the other two folds have a 90/60 split.
+
+```{r}
+lapply(p$pred$instance$train.inds, function(x) length(x))
+```
+
+This is caused by the fact that we supplied five groups that must belong together but only used a three fold resampling strategy here.
+
+### Blocking with fully pre-defined indices
+
+There is a second way of using pre-defined indices in resampling in `mlr`: Constructing the folds based on the supplied indices in `blocking`.
+We refer to this method here as "grouping" to distinguish it from "blocking".
+This method is more restrictive in the way that it will always use the number of levels supplied via `blocking` as the number of folds.
+To use this method, we need to set `fixed = TRUE` instead of `blocking.cv` when creating the resampling description object.
+
+We can leave out the `iters` argument, as it will be set internally to the number of supplied factor levels.
+
+```{r}
+rdesc = makeResampleDesc("CV", fixed = TRUE)
+p = resample("classif.lda", task, rdesc)
+sort(p$pred$instance$train.inds[[1]])
+```
+
+You can see that we automatically created five folds in which the test set always corresponds to one factor level.
+
+Doing it this way also means that we cannot do repeated CV because there is no way to create multiple shuffled folds of this fixed arrangement.
+
+```{r}
+lapply(p$pred$instance$train.inds, function(x) length(x))
+```
+
+However, this method can also be used in nested resampling settings (e.g. in hyperparameter tuning).
+In the inner level, the factor levels are honored and the function simply creates one fold less than in the outer level. 
+
+```{r, warning=FALSE}
+# test fixed in nested resampling
+lrn = makeLearner("classif.lda")
+ctrl <- makeTuneControlRandom(maxit = 2)
+ps <- makeParamSet(makeNumericParam("nu", lower = 2, upper = 20))
+inner = makeResampleDesc("CV", fixed = TRUE)
+outer = makeResampleDesc("CV", fixed = TRUE)
+tune_wrapper = makeTuneWrapper(lrn, resampling = inner, par.set = ps,
+  control = ctrl, show.info = FALSE)
+
+p = resample(tune_wrapper, task, outer, show.info = FALSE,
+  extract = getTuneResult)
+```
+
+To check on the inner resampling indices, you can call `getResamplingIndices(inner = TRUE)`.
+You can see that for every outer fold (List of 5), four inner folds were created that honor the grouping supplied via the `blocking` argument.
+
+Of course you can also use a normal random sampling "CV" description in the inner level by just setting `fixed = FALSE`.
+
+```{r}
+str(getResamplingIndices(p, inner = TRUE))
+```
+
```

Comment:
Could you please mention here that the number of inner folds is automatically adjusted based on the available levels?

Reply:
Yes, as said, I will write a new section comparing blocking and grouping and explain what happens.